lib/matplotlib/axes/_axes.py

```diff
@@ -5871,6 +5871,40 @@ def hist(self, x, bins=10, range=None, normed=False, weights=None,
         .. plot:: mpl_examples/statistics/histogram_demo_features.py
 
         """
+        def _normalize_input(inp, ename='input'):
+            """Normalize 1 or 2d input into list of np.ndarray or
+            a single 2D np.ndarray.
```

Comment:
it really does return either/or :imp: 

Note that this is a `_`'d private to this method function.  The main reason for this existing is to remove copy-paste code so the questionable API is OK (imo) as it preserves the behavior of the block of code the function call is replacing.